### ISSUE TITLE
Eval Mode in select()

### ISSUE THREAD

**Issue by nab170130** [decile-team/trust#3]
The select() methods for the targeted subset selection strategies need to have self.model.eval(). Without it, the embedding computation will not necessarily be the same for similar points, which may cause performance degradation.
